### PR TITLE
ddl: enable linter revive

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #40786

### What is changed and how it works?

Enable revive for `ddl` package.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: build/nogo_config.json [hunk 1]
```diff
@@ -525,6 +525,7 @@
       "executor/internal/": "executor/internal/",
       "types/json_binary_functions.go": "types/json_binary_functions.go",
       "types/json_binary_test.go": "types/json_binary_test.go",
+      "ddl/": "ddl",
       "ddl/backfilling.go": "ddl/backfilling.go",
       "ddl/column.go": "ddl/column.go",
       "ddl/index.go": "ddl/index.go",
```
Comment (hawkingrei) [inline build/nogo_config.json:531]: ```ddl/*``` in the revive can be deleted.

Comment (hawkingrei) [inline build/nogo_config.json:531]: ```ddl/*``` in the revive can be deleted.

**Review by hawkingrei — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**